angr/procedures/libc/time.py

```diff
@@ -0,0 +1,11 @@
+import angr
+import time as _time
+
+class time(angr.SimProcedure):
+    #pylint:disable=arguments-differ
+    def run(self, time_ptr):
+        ts = int(_time.time())
+        if self.state.solver.eval(time_ptr) != 0:
+            ts_bv = self.state.solver.BVV(ts, 64).reversed
+            self.state.memory.store(time_ptr, ts_bv)
```

Comment:
this will only work for little-endian architectures. use `endness=self.state.arch.memory_endness` in the store command instead of manually reversing the bitvector.

Furthermore, you should try to respect the `USE_SYSTEM_TIMES` state option, and return a symbolic value if it is not set.

Reply:
I miss the option and take care of it now, thanks for your notice!

Also I found the system call in `linux-kernel/time` procedure won't care about the `USE_SYSTEM_TIMES` option, so I move my implement there.